tests/collections/asr/decoding/test_fast_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,187 @@
+import glob
+import json
+import os
+import sys
+import tempfile
+import traceback
+
+import ipdb
+import jiwer
+import pytest
+import torch
+from omegaconf import OmegaConf, open_dict
+
+from nemo.collections.asr.models import ASRModel
+from nemo.collections.asr.models.rnnt_bpe_models import EncDecRNNTBPEModel
+from nemo.collections.asr.parts.submodules.fast_rnnt_greedy_decoding import RNNTGreedyDecodeFast
+from nemo.collections.asr.parts.submodules.rnnt_greedy_decoding import GreedyBatchedRNNTInfer
+
+
+@pytest.mark.parametrize(
+    ("model_name", "batch_size", "use_subset", "enable_bfloat16"),
+    [
+        # The xfail is not catching for some reason...
+        # pytest.param("stt_en_fastconformer_transducer_large", 8, True, True, marks=pytest.mark.xfail(reason="Cannot instantiate graph with persistent RNN")),
+        ("stt_en_fastconformer_transducer_large", 7, False, True),
+        ("stt_en_fastconformer_transducer_xlarge", 16, False, True),
+        ("stt_en_fastconformer_transducer_xlarge", 16, True, True),
+        ("stt_en_fastconformer_transducer_xlarge", 128, True, True),
+        ("stt_en_fastconformer_transducer_xlarge", 16, False, False),
+    ],
+)
+def test_for_loop(model_name, batch_size, use_subset, enable_bfloat16):
+    nemo_model = ASRModel.from_pretrained(model_name, map_location="cuda")
+    conf = nemo_model.to_config_dict()
+    with open_dict(conf):
+        conf["decoding"]["greedy"]["max_symbols"] = 5
+        conf["decoding"]["greedy"]["loop_labels"] = False
+
+    with tempfile.NamedTemporaryFile() as fp:
+        OmegaConf.save(config=conf, f=fp.name)
+        nemo_model = ASRModel.from_pretrained(model_name, override_config_path=fp.name, map_location="cuda")
+    nemo_model.freeze()
+
+    nemo_model.preprocessor.featurizer.dither = 0.0
+    nemo_model.preprocessor.featurizer.pad_to = 0
+
+    # Switch model to evaluation mode
+    nemo_model.eval()
+    # Freeze the encoder and decoder modules
+    nemo_model.encoder.freeze()
+    nemo_model.decoder.freeze()
+    nemo_model.joint.freeze()
+
+    audio_filepaths = glob.glob("/home/dgalvez/scratch/data/LibriSpeech/test-clean-processed/*.wav")
```

Comment:
@titu1994 I am looping over a dataset and making sure that the results are the same between the old implementation and the new one. Does that work for you as a test?

Secondly, how do you actually run tests on datasets? I'm depending on a local path on my machine right now, which is obviously not okay.

Reply:
We don't run tests on datasets usually. We do have a few samples of AN4 inside the NeMo test dir, could you use that instead of hard coded dataset path? We only need to check if the text is somewhat similar over 4-5 files. 

See examples of how to access the test_data_dir pytest fixture and relative path to the AN4 files inside the ASR tests (I think the dataset ones specifically).

Reply:
As to how to run with datasets, cd to root of NeMo dir and run pytest tests/asr/ with any additional flags you want usually --relax_numba_compat is necessary

Reply:
Sounds good.